client/components/tinymce/plugins/wpcom-view/wpvideo-view-container.jsx

```diff
@@ -0,0 +1,88 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import shortcodeUtils from 'lib/shortcode';
+import MediaActions from 'lib/media/actions';
+import MediaListStore from 'lib/media/list-store';
+import WpVideoView from './wpvideo-view';
+
+const getStateData = siteId => {
+	return {
+		media: MediaListStore.getAll( siteId ),
+		mediaHasNextPage: MediaListStore.hasNextPage( siteId ),
+		mediaFetchingNextPage: MediaListStore.isFetchingNextPage( siteId )
+	};
+};
+
+class WpVideoViewContainer extends Component {
+
+	static match( content ) {
+		const match = shortcodeUtils.next( 'wpvideo', content );
+
+		if ( match ) {
+			return {
+				index: match.index,
+				content: match.content,
+				options: {
+					shortcode: match.shortcode
+				}
+			};
+		}
+	}
+
+	static serialize( content ) {
+		return encodeURIComponent( content );
+	}
+
+	constructor( props ) {
+		super( props );
+		this.state = getStateData( props.siteId );
+	}
+
+	componentDidMount() {
+		MediaListStore.on( 'change', this.updateStateData.bind( this ) );
+		this.fetchData( this.props.siteId );
+	}
+
+	componentWillUnmount() {
+		MediaListStore.off( 'change', this.updateStateData.bind( this ) );
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if ( this.props.siteId !== nextProps.siteId ) {
+			this.fetchData( this.props.siteId );
+			this.updateStateData();
+		}
+	}
+
+	fetchData( siteId ) {
+		setTimeout( () => {
```

Comment:
So, from what I can tell, the purpose for querying/fetching media is to determine the dimensions for the embedded video? Can we not just default this to a fixed width value, similar to [what we're doing for images](https://github.com/Automattic/wp-calypso/blob/fb83eac4d214d85f978225172a57080da3e0996d/client/components/tinymce/plugins/media/restrict-size/index.js#L13)? To be clear, I'm not totally familiar with proportion variations for VideoPress videos, or to the exact extent `videopress-iframe.js` is capable of managing these details for us.